byway/src/main/java/com/google/sps/data/Trip.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+
+/**
+ * A class to make a Trip type, containing
+ * a specific id, destinations, interests and routes 
+ * to be used when updating a trip.
+ */
+public final class Trip {
+
+  private String id;
+  private String start;
+  private ArrayList<String> destinations;
+  private ArrayList<String> interests;
+  private ArrayList<String> route;
+
+  public Trip(String id, String start, ArrayList<String> destinations,
+              ArrayList<String> interests, ArrayList<String> route) {
+    this.id = id;
+    this.start = start;
+    this.destinations = destinations;
+    this.interests = interests;
+    this.route = route;
```

Comment:
Note that when writing container types it's often a bad idea to blindly trust that a collection provided by calling code won't be used elsewhere. Sharing the same list in different places by accident can lead to unintended issues, such as accidentally adding things to the same list multiple times. You may want to create a new list with a copy of the contents.

Note that if you use `this.interests = new ArrayList<>(interests)`, you can be more general in what type the constructor accepts, since you can construct an arraylist by [copying any `Collection`](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#ArrayList-java.util.Collection-), you can make the arguments `Collection<T>` instead of `ArrayList<T>`, which gives the caller more choice of what type to pass in.

In general, it's a good idea to make the types of arguments you accept as general as reasonable, to provide the calling code with maximum flexibility around how they use your code.

Reply:
If it were modified elsewhere, after making a Trip class, wouldn't it leave the Trip containers unmodified since it passes in a copy? This makes it sound as if these collections would be passed by reference, which Java doesn't do. Maybe I'm wrong, since said parameters being passed would require a new keyword to make an object, which would likely be using the same object (in which case what you said would apply). Not sure how Java works internally so would appreciate some insight/clarification.

Also, fixed